interop.go

```diff
@@ -0,0 +1,139 @@
+package pglock
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+func FromSQLDB(db *sql.DB) *sqldbWrapper {
+	return &sqldbWrapper{
+		DB: db,
+	}
+}
+
+var (
+	_ sqlLike = (*sqldbWrapper)(nil)
+	_ txLike  = (*sqlTxWrapper)(nil)
+)
+
+type sqldbWrapper struct {
+	*sql.DB
+}
+
+func (s *sqldbWrapper) BeginTx(ctx context.Context, opts *sql.TxOptions) (txLike, error) {
+	t, err := s.DB.BeginTx(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	return &sqlTxWrapper{Tx: t}, nil
+}
+
+func (s *sqldbWrapper) Exec(sql string, arguments ...any) (any, error) {
+	return s.DB.Exec(sql, arguments...)
+}
+
+func (s *sqldbWrapper) QueryRowContext(ctx context.Context, query string, args ...any) rowLike {
+	return s.DB.QueryRowContext(ctx, query, args...)
+}
+
+func (s *sqldbWrapper) QueryContext(ctx context.Context, query string, args ...any) (rowsLike, error) {
+	return s.DB.QueryContext(ctx, query, args...)
+}
+
+type sqlTxWrapper struct {
+	*sql.Tx
+}
+
+func (t *sqlTxWrapper) ExecContext(ctx context.Context, sql string, arguments ...any) (resultLike, error) {
+	return t.Tx.ExecContext(ctx, sql, arguments...)
```

Comment:
I really wonder why you need to define a custom wrapper in the first place.
At the very least, you could just use alias